contrib/inventory/ec2.py

```diff
@@ -590,7 +590,7 @@ def get_rds_instances_by_region(self, region):
                 error = self.get_auth_error_message()
             if not e.reason == "Forbidden":
                 error = "Looks like AWS RDS is down:\n%s" % e.message
-            self.fail_with_error(error, 'getting RDS instances')
+                self.fail_with_error(error, 'getting RDS instances')
```

Comment:
well, the whole error block should `self.fail_with_error` - that is why it is not nested in the IF originally - by placing it in the IF here, setting the error code on L590 does not make it back to the user